test/withWasm.spec.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import Wasm from '../src';
+import withWasm from '../src/withWasm';
+
+jest.mock('../src', () => ({
+  __esModule: true,
+  default: jest.fn(({ children }) => (
+    <div>
+      {children({
+        loading: true,
+        error: false,
+        data: { hello: 'world' }
+      })}
+    </div>
+  ))
+}));
+
+function Component({ firstname }) {
+  return <div>Hello {firstname}</div>;
+}
+
+describe('withWasm ', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    const Enhanced = withWasm(Component);
+
+    const instance = TestRenderer.create(
+      <Enhanced firstname="James" url="/add.wasm" />
```

Comment:
thank you for time and effort that you are putting on this 😄

I think that the child component should not receive `url`,` bufferSource` and `importObject`. `importObject` might be useful to access the module memory, however it can be also exported from the module itself and accessible through the `instance` object. I personally think that second way is cleaner, we have a single object that holds all the information and we encourage encapsulation.
For example, if we decide to use two modules that both import the memory from the importObject:
- module A exports the memory into the instance object
- module B doesn't want to export the memory, it want to be the only one that uses it

If we decide to provide the importObject to the child component, it will be able to access the memory of both, so, to avoid this for the module B we have to create another HoC that removes the `importObject` prop.
However, if a js module want to export the `importObject` it can done that using and `export const` or a factory...
These are my thoughts, what do you think about it?